File: DI_IOC_Observer/DI_IOC_Observer_JAVA/src/com/kretzer/Hexagon.java
package com.kretzer;

import com.sun.javaws.exceptions.InvalidArgumentException;

/**
 * Created by jason for DI_IOC_Observer_JAVA.
 */
public class Hexagon extends AbstractPolygon {
    public Hexagon(double initSideLength) throws InvalidArgumentException {
        sides = 6;
        setSideLength(initSideLength);
    }

}

File: DI_IOC_Observer/DI_IOC_Observer_CS/DI_IOC_Observer_CS/Hexagon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DI_IOC_Observer_CS
{
    class Hexagon : AbstractPolygon
    {
        public Hexagon(double initSideLength) {
            sides = 6;
            setSideLength(initSideLength);
        }
    }
}
